src/lib.rs

```diff
@@ -352,6 +352,93 @@ impl Filter {
     }
 }
 
+/// The possible input methods.
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile(String),
+
+    /// Read from `stdin`.
+    Stdin,
+}
+
+impl InputMethod {
+    /// Create a new instance from the given command line options.
+    pub fn parse(matches: &getopts::Matches) -> Option<Self> {
+        let mut method = 0;
+
+        if matches.opt_present("git") {
+            method |= 1
+        }
+
+        if matches.opt_present("input") {
+            method |= 2
+        }
+
+        if matches.opt_present("stdin") {
+            method |= 4
+        }
+
+        let method = method;
```

Comment:
Hmm, I would remove this, the code below is very short and it is improbable, that it is changed by accident there.

Reply:
I would rather leave it in, just to be sure.

Reply:
Hmm, I think it bloats the code a bit, since it is not necessary and makes the function as a whole less overseeable. 😅

Reply:
Okay, I can combine the reversion with the changes due to your suggestion for the implementation of the different return type tomorrow. :wrench: